### PR TITLE
Add DataFrame arithmetics

### PR DESCRIPTION
**Pull request by sinhrks**
This PR allows elementwise ops for ``DataFrame``.

```
import pandas as pd
import dask.dataframe as dd
df = dd.from_pandas(pd.DataFrame({'x': [1, 2, 3, 4], 'y': [5, 6, 7, 8]}), 2)
df.compute()
#    x  y
# 0  1  5
# 1  2  6
# 2  3  7
# 3  4  8

(df + 1).compute()
#    x  y
# 0  2  6
# 1  3  7
# 2  4  8
# 3  5  9

(df + df).compute()
#    x   y
# 0  2  10
# 1  4  12
# 2  6  14
# 3  8  16
```

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Will this be extra expensive if `df1` and `df2` are have different divisions along the index? It may be worth adding a note to [this](http://dask.pydata.org/en/latest/dataframe.html#what-definitely-works) section of the documentation, if my intuition is correct.

Also, I love that all the pandas devs are interested in / contributing to dask :)

File: dask/dataframe/tests/test_dataframe.py [hunk 1]
```diff
@@ -187,6 +188,113 @@ def test_arithmetic():
...
+    tm.assert_frame_equal((d2 % d3).compute(), full2 % full3)
+    tm.assert_frame_equal((d2 > d3).compute(), full2 > full3)
+    tm.assert_frame_equal((d2 < d3).compute(), full2 < full3)
+    tm.assert_frame_equal((d2 >= d3).compute(), full2 >= full3)
+    tm.assert_frame_equal((d2 <= d3).compute(), full2 <= full3)
+    tm.assert_frame_equal((d2 == d3).compute(), full2 == full3)
+    tm.assert_frame_equal((d2 != d3).compute(), full2 != full3)
+
+    """
+    dsk4 = {('x', 0): pd.DataFrame({'a': [3, 2, 1], 'b': [7, 8, 9]},
```
Comment (sinhrks) [inline dask/dataframe/tests/test_dataframe.py]: Ah, found a problem in test case, I'll dig in..

**Comment by mrocklin** [comment]
We have logic to handle differently partitioned dataframes in `multi.py::align_partitions`.  This is used to handle merges along the index but should also be good for elementwise operations along imperfectly aligned dataframes.

**Comment by mrocklin** [comment]
BTW, I'm really happy to see things like this, especially coming from Pandas developers.  My knowledge of the Pandas codebase is very light and I'm happy to see some more experienced Pandas developers come in to dask.dataframe.  If there are some core decisions that seem wrong to you please speak up and lets change them.

**Comment by sinhrks** [comment]
@TomAugspurger, @mrocklin Thanks, I think it should work now for same divisions.

I'll take a look for ``align_partitions``, but can it be handled separatly?

**Comment by mrocklin** [comment]
This looks good to me.  Merging soon if no comment.  

I agree that we can handle `align_partitions` separately.  I'll take a look at this early this week if no one else gets to it first.

**Pull request merged**